File: src/main/java/artisra/topinterview150/linkedlist/ListNode.java
package artisra.topinterview150.linkedlist;

/**
 * Represents a node in a singly linked list.
 * <p>
 * Each node contains an integer value and a reference to the next node in the list.
 */
public class ListNode {
    public int val;
    public ListNode next;

    public ListNode(int x) {
        val = x;
        next = null;
    }

    public ListNode(int val, ListNode next) {
        this.val = val;
        this.next = next;
    }

    @Override
    public String toString() {
        return "ListNode(value=" + val + ",next=" + next + ")";
    }
}

File: src/main/kotlin/topinterview150/linkedlist/ListNode.kt
package dev.artisra.topinterview150.linkedlist

/**
 * Represents a node in a linked list.
 */
class ListNode(var value: Int) {
    var next: ListNode? = null
    override fun toString(): String {
        return "ListNode(value=$value,next=$next)"
    }
}